### DISCUSSION TITLE
How can we use migration on Fastapi?

### DISCUSSION THREAD

**Discussion by shadiqurrahaman** [fastapi/fastapi#4125] [Questions] [answered]
I want a migration feature with fastapi. how can we achieve it?

**Comment by simondale00** [comment]
Hey @shadiqurrahaman. Can you expand on the ask a little more? What are you trying to migrate, and to where?

Best,
Simon

**Reply by shadiqurrahaman** [reply]
I want to keep track all my database table's collumn name and data types locally in python file

**Reply by simondale00** [reply, marked as answer]
I've used [Alembic](https://alembic.sqlalchemy.org/en/latest/) with success in the past. It works well if you're using SQLAlchemy. It also keeps all your migrations/data mappers in native python.

I've also used [DBmate](https://github.com/amacneil/dbmate) with some success in the past. It's framework agnostic, allowing you to define migrations in raw SQL. The big advantage is that you're not tied to a specific language or framework.

Good luck!

Simon
